internals/metrics/metrics_test.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2025 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package metrics_test
+
+import (
+	"bytes"
+	"testing"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/canonical/pebble/internals/metrics"
+)
+
+func Test(t *testing.T) {
+	TestingT(t)
+}
+
+type OpenTelemetryWriterSuite struct{}
+
+var _ = Suite(&OpenTelemetryWriterSuite{})
+
+func (s *OpenTelemetryWriterSuite) TestOpenTelemetryWriter(c *C) {
```

Comment:
Nit: because this is in `OpenTelemetryWriterSuite`, the test name can just be `TestWriter`:
```suggestion
func (s *OpenTelemetryWriterSuite) TestWriter(c *C) {
```